server/src/main/java/org/elasticsearch/index/mapper/DocValueFetcher.java

```diff
@@ -8,46 +8,27 @@
 
 package org.elasticsearch.index.mapper;
 
-import org.apache.lucene.index.LeafReaderContext;
-import org.elasticsearch.index.fielddata.FormattedDocValues;
-import org.elasticsearch.index.fielddata.IndexFieldData;
 import org.elasticsearch.search.DocValueFormat;
-import org.elasticsearch.search.lookup.SourceLookup;
+import org.elasticsearch.search.lookup.ValuesLookup;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 
-import static java.util.Collections.emptyList;
-
 /**
  * Value fetcher that loads from doc values.
  */
 public final class DocValueFetcher implements ValueFetcher {
```

Comment:
Essentially all of this class is now moved into ValuesLookup.docValues() so it might be worth just removing it and replacing it with lambdas wherever it is used.